auth-lib/src/auth/java/com/spotify/sdk/android/auth/browser/LoginDialog.java

```diff
@@ -202,8 +203,9 @@ public void onDetachedFromWindow() {
 
     @Override
     protected void onStop() {
-        if (!mResultDelivered && mListener != null) {
-            mListener.onCancel();
+        AuthorizationHandler.OnCompleteListener listener = mListener;
+        if (!mResultDelivered && listener != null) {
+            listener.onCancel();
```

Comment:
Why do we need a local variable for the listener here?

Reply:
To be sure the listener wasn't nullified between the moment we check it and call it, it could happen if the setter for the listener is being called from another thread